samples/snippets/src/main/java/com/example/bigquerystorage/WritePendingStream.java

```diff
@@ -48,63 +60,124 @@ public static void runWritePendingStream()
+    if (commitResponse.hasCommitTime() == false) {
+      for (StorageError err : commitResponse.getStreamErrorsList()) {
+        System.out.println(err.getErrorMessage());
+      }
+      throw new RuntimeException("Error committing the streams");
+    }
+    System.out.println("Appended and committed records successfully.");
+  }
+
+  public static WritePendingStream create(TableName parentTable, BigQueryWriteClient client)
+      throws IOException, DescriptorValidationException, InterruptedException {
+    WritePendingStream writer = new WritePendingStream();
+    writer.client = client;
+
+    // Initialize a write stream for the specified table.
+    // For more information on WriteStream.Type, see:
+    // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1/WriteStream.Type.html
+    WriteStream stream = WriteStream.newBuilder().setType(WriteStream.Type.PENDING).build();
+
+    CreateWriteStreamRequest createWriteStreamRequest =
+        CreateWriteStreamRequest.newBuilder()
+            .setParent(parentTable.toString())
+            .setWriteStream(stream)
+            .build();
+    writer.writeStream = client.createWriteStream(createWriteStreamRequest);
+
+    // Use the JSON stream writer to send records in JSON format.
+    // For more information about JsonStreamWriter, see:
+    // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1beta2/JsonStreamWriter.html
+    writer.streamWriter =
+        JsonStreamWriter.newBuilder(
+                writer.writeStream.getName(), writer.writeStream.getTableSchema())
+            .build();
+    return writer;
+  }
+
+  public void append(JSONArray data)
+      throws DescriptorValidationException, IOException, ExecutionException {
+    ApiFuture<AppendRowsResponse> future = streamWriter.append(data);
+    this.inflightRequestPhaser.register();
+    ApiFutures.addCallback(
+        future, new AppendCompleteCallback(this), MoreExecutors.directExecutor());
+  }
+
+  public void cleanup() {
+    // Wait for all in-flight requests to complete.
+    this.inflightRequestPhaser.arriveAndAwaitAdvance();
+    FinalizeWriteStreamResponse finalizeResponse =
+        client.finalizeWriteStream(writeStream.getName());
+    System.out.println("Rows written: " + finalizeResponse.getRowCount());
+  }
+
+  public String getStreamName() {
+    return this.writeStream.getName();
+  }
+
+  static class AppendCompleteCallback implements ApiFutureCallback<AppendRowsResponse> {
+
+    private final WritePendingStream parent;
+
+    AppendCompleteCallback(WritePendingStream parent) {
+      this.parent = parent;
+    }
+
+    public void onSuccess(AppendRowsResponse response) {
+      if (response.hasError()) {
+        System.out.format("Error: %s\n", response.getError());
+      } else {
+        System.out.format("Append %d success\n", response.getAppendResult().getOffset().getValue());
```

Comment:
For pending stream, show user a way to set stream_status_ and append to check on the status to fail the append?

Reply:
Done, thought I'm not great at Java, so maybe there's a better way?